packages/loader/container-loader/src/container.ts

```diff
@@ -2432,10 +2442,39 @@ export class Container
 		return { snapshot, versionId: version?.id };
 	}
 
+	private async fetchSnapshot(
+		specifiedVersion: string | undefined,
+	): Promise<{ snapshot?: ISnapshot | ISnapshotTree; versionId?: string }> {
+		if (
+			this.mc.config.getBoolean("Fluid.Container.FetchSnapshotInNewFormat") === true &&
+			this.service?.policies?.supportNewSnapshotFormat === true
+		) {
+			const snapshot = await this.storageAdapter.getSnapshot({
+				id: specifiedVersion ?? "",
+				treeId: specifiedVersion ?? "",
+			});
+			const version: IVersion = {
```

Comment:
nit - we only use version.id, so do not need version structure.

Reply:
Yes, in the api, I am only passing the string but we have a prop on container, loadedFromVersions which is Iversion, type, so we need it here to be in that form.